aws/rust-runtime/aws-inlineable/src/s3_express.rs

```diff
@@ -727,3 +727,124 @@ pub(crate) mod identity_provider {
         }
     }
 }
+
+/// Supporting code for S3 Express runtime plugin
+pub(crate) mod runtime_plugin {
+    use aws_smithy_runtime_api::client::runtime_plugin::RuntimePlugin;
+    use aws_smithy_types::config_bag::{CloneableLayer, FrozenLayer, Layer};
+    use aws_types::os_shim_internal::Env;
+
+    mod env {
+        pub(super) const S3_DISABLE_EXPRESS_SESSION_AUTH: &str =
+            "AWS_S3_DISABLE_EXPRESS_SESSION_AUTH";
+    }
+
+    #[derive(Debug)]
+    pub(crate) struct S3ExpressRuntimePlugin {
+        config: FrozenLayer,
+    }
+
+    impl S3ExpressRuntimePlugin {
+        pub(crate) fn new(service_config: &CloneableLayer) -> Self {
+            Self::new_with(service_config, Env::real())
+        }
+
+        fn new_with(service_config: &CloneableLayer, env: Env) -> Self {
+            let mut layer = Layer::new("S3ExpressRuntimePlugin");
+            if service_config
+                .load::<crate::config::DisableS3ExpressSessionAuth>()
+                .is_none()
+            {
+                match env.get(env::S3_DISABLE_EXPRESS_SESSION_AUTH) {
+                    Ok(value)
+                        if value.eq_ignore_ascii_case("true")
+                            || value.eq_ignore_ascii_case("false") =>
+                    {
```

Comment:
Is the expected behavior that the value is just ignored if it's not true or false? E.g., `S3_DISABLE_EXPRESS_SESSION_AUTH=1` would result in it being ignored? Is it supposed to warn in that case?